posthog/models/dashboard_tile.py

```diff
@@ -25,11 +42,38 @@ class DashboardTile(models.Model):
 
     class Meta:
         indexes = [models.Index(fields=["filters_hash"], name="query_by_filters_hash_idx")]
+        constraints = [
+            UniqueConstraint(
+                fields=["dashboard", "insight"],
+                name=f"unique_dashboard_insight",
+                condition=Q(("insight__isnull", False)),
+            ),
+            UniqueConstraint(
+                fields=["dashboard", "text"], name=f"unique_dashboard_text", condition=Q(("text__isnull", False))
+            ),
+            models.CheckConstraint(check=build_check(("insight", "text")), name="dash_tile_exactly_one_related_object"),
+        ]
+
+    def clean(self):
+        super().clean()
+
+        related_fields = sum(map(bool, [getattr(self, o_field) for o_field in ("insight", "text")]))
+        if related_fields != 1:
+            raise ValidationError("Can only set either an insight or a text for this tile")
```

Comment:
Is there a test for this case?

Reply:
test: https://github.com/PostHog/posthog/pull/12110/files#diff-4eb35ff7247f20c2e6e8a8c16c605c734d9c9a2cbe078671e00717ab6ae8e1e5R41